controllers/component_controller.go

```diff
@@ -253,8 +253,22 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
+	if component.Spec.Source.ImageSource != nil {
```

Comment:
So I am not sure if our understanding of the `Component` spec is on the same page here. I took a glance at the implementation and it seems like when `component.Spec.Source.ImageSource` is specified, we build from `component.Spec.Source.GitSource.URL` and output it to the image `component.Spec.Source.ImageSource` 

But `component.Spec.Source.ImageSource` & `component.Spec.Source.GitSource.URL` are two kinds of src. A user on HAC during import; can choose to import from devfile sample/their own git project or specify an image. This image is represented as `component.Spec.Source.ImageSource`, so its an Union.

If I'm right, we are interested in a devfile's [component.image.imageName](https://github.com/nodeshift-starters/devfile-sample/blob/main/devfile.yaml#L17) which holds the output where image build needs to go to

Reply:
Ah I see, I wasn't aware of that.

Reply:
Either way, the 'output' image needs to be presented by the user - is what I think.

Reply:
Interesting, so will we be using an output image for every build? I was of the opinion that it would be only be for the outerloop 🤔

Reply:
A couple of maybe-silly questions from the spec:
```
spec:
  componentName: backend
  application: application-sample
  source:
    git:
      url: https://github.com/sbose78/devfile-sample-code-with-quarkus
    image:
      containerImage: docker.io/sbose78/foobar:dev-code-with-quarkus
```
Is "componentName" provided by the end user?

What is the foobar here and how does it get set?

I see @maysunfaisal referring to component.Spec.Source.ImageSource, component.Spec.Source.GitSource.URL,  component.Spec.Source.ImageSource, and component.image.imageName.  How does those map to the field names in your PR comment: component.SpecSource.Git.URL and component.Spec.Source.Image.ContainerImage?

Reply:
@amfred I can answer few of these ques here

> I have a few maybe-silly questions from the spec in your PR comment, @sbose78 :
> 
> ```
> spec:
>   componentName: backend
>   application: application-sample
>   source:
>     git:
>       url: https://github.com/sbose78/devfile-sample-code-with-quarkus
>     image:
>       containerImage: docker.io/sbose78/foobar:dev-code-with-quarkus
> ```
> 
> Is "componentName" provided by the end user?
> 

I think yes, when a user adds the component via HAC, they would probably have a way to specify the component name that is being added.

> What is the foobar here and how does it get set?
> 

I think this is just the image name that is being called here in this example, it probably should be backend to reflect the component in question(probably namespaced to handle it better)

> I see @maysunfaisal referring to component.Spec.Source.ImageSource, component.Spec.Source.GitSource.URL, component.Spec.Source.ImageSource, and component.image.imageName. How does those map to the field names in your PR comment: component.SpecSource.Git.URL and component.Spec.Source.Image.ContainerImage?

HAC imports cases
1. devfiile sample/git - HAC uses component.Spec.Source.GitSource.URL
2. container image - HAC uses component.Spec.Source.ImageSource

In use case 1, the devfile that is associated will have a component component[i].image.imageName which can be either preset/input via HAC by the user. And this will store the build output.

In use case 2, since its only a container image(String) thats being inputted, a basic devfile would be created with the image specified. Not entirely sure how the builds would work in this case.